packages/@aws-cdk/aws-codecommit/lib/repository.ts

```diff
@@ -488,6 +489,31 @@ export interface RepositoryProps {
    * @default - No description.
    */
   readonly description?: string;
+
+  /**
+   * The asset the repository with the given branch should be initialized
+   *
+   * @default - No initialization (create empty repo)
+   */
+  readonly code?: CodeInitializationProps;
```

Comment:
I don't love this API. In my opinion, it's too low-level.

I think a better design would be to introduce a `Code` class, similar to the `Code` class from the Lambda module. This way, you can have static factory methods on the `Code` class, and they can have different ways to pass this information to the construct. The Asset will be created by the implementation, not by the customer of this API.

So, I'm looking at an experience like:

```ts
const repo = new codecommit.Repository(this, 'Repository', {
    repositoryName: 'MyRepositoryName',
    code: codecommit.Code.fromDirectory('src/code', 'main'), // branch name is the optional second argument
});
```

Reply:
okay, I tried to implement it :)